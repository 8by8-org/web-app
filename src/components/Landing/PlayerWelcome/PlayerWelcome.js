import React, { useState, useEffect } from "react";
import { useHistory } from "react-router-dom";
import { getFirestore, getDoc, doc } from "firebase/firestore";
import SignUp1 from "../../../assets/images/PlayerWelcome/SignUp1.png";
import SignUp2 from "../../../assets/images/PlayerWelcome/SignUp2.png";
import Vote from "../../../assets/images/PlayerWelcome/Vote1.png";
import BlackCurve from "../../../assets/images/PlayerWelcome/BlackCurve.svg";
import "./PlayerWelcome.scss";
import { auth } from "../../../firebase";
import { dummyPassword } from "../../../constants";
import { useAuth } from "../../../contexts/AuthContext";
import { LoadingWheel } from "./../../Utility/LoadingWheel/LoadingWheel.component";
import StepFour from "../../../assets/images/ChallengerWelcome/StepFour.png";

export default function PlayerWelcome({ isShare }) {
  const history = useHistory();
  const { currentUser } = useAuth();
  const [loading, setLoading] = useState(true);
  const [challengerInfo, setChallengerInfo] = useState(null);

  const url = window.location.href;
  const code = url.split("/").pop();

  async function getChallengerInfo() {
    const db = getFirestore();
    // If isShare is false then use the uid of the current user, else use the code gotten from the url.
    const docRef = doc(db, "users", isShare === false ? currentUser.uid : code);

    //only log in with substitute user if not already authenticated
    //only challenger's name and avatar is stored for security
    if (currentUser) {
      const query = await getDoc(docRef);
      // Just in case query.data() does not return anything, then send the user to the signin page.
      if (query.data()) {
        const info = (({ name, avatar }) => ({ name, avatar }))(query.data());
        info.challengerID = code;
        isShare === undefined &&
          localStorage.setItem("challengerInfo", JSON.stringify(info));
        if (isShare === false) {
          setChallengerInfo(info);
        } else {
          setChallengerInfo(JSON.parse(localStorage.getItem("challengerInfo")));
        }
      } else {
        history.push(`/signin`);
      }
    } else {
      await auth.signInWithEmailAndPassword(
        auth.getAuth(),
        "wkvxgesiknosbamhei@kvhrr.com",
        dummyPassword
      );
      const query = await getDoc(docRef);
      const info = (({ name, avatar }) => ({ name, avatar }))(query.data());
      info.challengerID = code;
      localStorage.setItem("challengerInfo", JSON.stringify(info));
      auth.getAuth().signOut();
      setChallengerInfo(JSON.parse(localStorage.getItem("challengerInfo")));
    }
  }

  // If code that is gotten from the url is playerwelcome or isShare is true then, if there is challengerInfo in
  // local storage then set it to challengerInfo, else send the user to the signin page. For eveything else run getChallengerInfo.
  useEffect(() => {
    code === "playerwelcome" || isShare
      ? localStorage.getItem("challengerInfo")
        ? setChallengerInfo(JSON.parse(localStorage.getItem("challengerInfo")))
        : history.push(`/signin`)
      : getChallengerInfo();
  }, []);

  // Render page after challengerInfo is gotten.
  useEffect(() => {
    challengerInfo && setLoading(false);
  }, [challengerInfo]);

  // If isShare is undefined, when playerwelcome page is rendered not in invite or share, then the buttons work.
  return loading === false ? (
    <div className={"player-welcome" + ((isShare || isShare === false) ? " preview" : "")}>
      <div className="top" style={{textAlign : "center"}}>
        <h1 className="top-heading">
          <span className="underline h1">Support</span>{" "}
          {challengerInfo && challengerInfo.name !== null
            ? `${challengerInfo.name}'s`
            : "the"}{" "}
          8by8 Challenge!
        </h1>

        <p className="text">
          <b>
            Help{" "}
            {challengerInfo && challengerInfo.name !== null
              ? challengerInfo.name
              : "your friend"}{" "}
            win their <u>8by8 Challenge</u> by registering to vote or taking
            other actions to #stopasianhate!
          </b>
        </p>

        <button
          className="gradient"
          onClick={() => {
            isShare === undefined && history.push(`/actions`);
          }}
        >
          <span>Get Started</span>
        </button>
        {!currentUser && (
          <div align="center">
            <p className="small-text">
              Already have an account?{" "}
              <button
                className="signin-link blue"
                onClick={() => {
                  isShare === undefined && history.push(`/signin`);
                }}
              >
                Sign In
              </button>
            </p>
          </div>
        )}
        <p>
          <a href="/why8by8" className="link" style={{ color : "#02DDC3", zIndex: "1"}}> See why others are doing it</a>
          </p>
      </div>
      <img src={BlackCurve} className="curve" alt="Black Curve" />
      <div className="main-content">
        <h2 className="heading underline">Here's how it works</h2>
        <h3>1. Choose an action to take</h3>
        <p className="text">
          You can take any number of the available actions: register to vote,
          get election reminders or take the 8by8 challenge yourself. Pick one
          to start.
        </p>
        <div className="image">
          <img src={SignUp1} alt="8by8 Logo" />
        </div>
        <h3>2. Your friend will earn a badge</h3>
        <p className="text">
          Any of the 3 actions will help your friend earn a badge and get closer
          to winning the challenge.
        </p>
        <div className="image">
          <img src={SignUp2} alt="8by8 Logo" />
        </div>
        <h3>3. Come back and take more actions</h3>
        <p className="text">
          Whether it is to help the same friend or a different one, the more
          actions you take, the better! Note that you can only help earn one
          badge per friend.
        </p>
        <div className="image">
          <img src={Vote} alt="8by8 Logo" />
        </div>
        <h3>4. Get a Reward!</h3>
        <p className="text">
        When your friend wins the challenge within 8 days, 
        you win as well! Then select and enjoy a reward 
        from one of our amazing partners.
        </p>
        <img
          src={StepFour}
          alt="earn 8 badges in 8 days"
          className="center-img"
        />
        <button
          className="gradient"
          onClick={() => {
            isShare === undefined && history.push(`/actions`);
          }}
        >
          <span>Get Started</span>
        </button>
        {!currentUser && (
          <p align="center" className="small-text">
            Already have an account?{" "}
            <button
              className="signin-link black"
              onClick={() => {
                isShare === undefined && history.push(`/signin`);
              }}
            >
              Sign In
            </button>
          </p>
        )}
      </div>
    </div>
  ) : (
    <LoadingWheel overlay={false} />
  );
}