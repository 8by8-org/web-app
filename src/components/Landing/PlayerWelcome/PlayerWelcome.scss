@import "src/styles/Components";

.preview {
  font-size: 12px;
}

.player-welcome {
  background-color: $white;
  color: $black;
  display: flex;
  flex-direction: column;

  .getStarted-button {
    @include button;
    outline: 4px solid white;
    margin-top: 25px;
  }

  .top {
    display: flex;
    flex-direction: column;
    padding: 8%;
    background-color: $black;
    color: white;
    .top-heading {
      @include heading-text(white, 312.5%);
      line-height: 115%;
      letter-spacing: 0.03em;
      margin: 20px 0px;
      .underline {
        @include text-underline;
        font-size: 100%;
        line-height: 115%;
        letter-spacing: 0.03em;
      }
    }
  }

  .signin-link {
    background-color: transparent;
    border: none;
    cursor: pointer;
    text-decoration: underline;
    display: inline;
    margin: 0;
    padding: 0px 0px 10px 0px;
    font-family: $lato;
    font-size: 94%;
    line-height: 22px;
    letter-spacing: 0.02em;
    font-weight: 900;
  }

  .blue {
    color: $blue;
  }

  .black {
    color: $black;
  }

  .main-content {
    padding: 40px 8%;

    h3 {
      align-self: flex-start;
      font-size: 150%;
    }
  }

  .heading {
    @include heading-text($black, 225%);
    margin-bottom: 40px;
    line-height: 43.2px;
    letter-spacing: 0.03em;
    text-align: center;
  }

  .text {
    font-family: $lato;
    font-size: 113%;
    line-height: 120%;
    letter-spacing: 0.02em;
    margin-bottom: 32px;
  }

  .small-text {
    font-family: $lato;
    font-size: 94%;
    letter-spacing: 0.02em;
    margin-top: 15px;
  }

  .curve {
    width: 100%;
  }

  .image {
    padding: 20px 0px;
    padding-bottom: 60px;
    text-align: center;
  }
}
