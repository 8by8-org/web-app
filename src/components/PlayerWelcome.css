@import url("https://fonts.googleapis.com/css2?family=Bebas+Neue&family=Lato&display=swap");

h1 {
  font-family: "Bebas Neue";
}

p {
  font-family: "Lato";
}

#button-style {
  color: black;
  background: #bfbfbf;
  border-radius: 0;
  border-color: #bfbfbf;
}
