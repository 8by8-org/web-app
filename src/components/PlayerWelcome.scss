@import "../styles/Components.scss";

.player-welcome {
  background-color: $white;
  color: $black;
  display: flex;
  flex-direction: column;

  .getStarted-button {
    @include button;
    outline: 4px solid white;
    margin: 10px;
  }

  .top {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    padding: 25px;
    background-color: $black;
    color: white;
    .top-heading {
      @include heading-text(white, 50px);
      line-height: 57.6px;
      letter-spacing: 0.03em;
      margin: 20px 0px 20px;
      .underline {
        @include text-underline;
      }
    }
  }

  .signin-link {
    background-color: transparent;
    border: none;
    cursor: pointer;
    text-decoration: underline;
    display: inline;
    margin: 0;
    padding: 0px 0px 10px 0px;
    font-family: $lato;
    font-size: 14px;
    line-height: 22px;
    letter-spacing: 0.02em;
    font-weight: 900;
  }

  .blue {
    color: $blue;
  }

  .black {
    color: $black;
  }

  .main-content {
    padding: 20px 40px;
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .heading {
    @include heading-text($black, 36px);
    margin-top: 25px;
    margin-bottom: 20px;
    line-height: 43.2px;
    letter-spacing: 0.03em;
    text-align: center;
    .underline {
      @include text-underline;
    }
  }

  .subheading {
    @include subheading-text($black, 24px);
    line-height: 29px;
    padding: 0px;
  }

  .text {
    font-family: $lato;
    font-size: 18px;
    line-height: 22px;
    letter-spacing: 0.02em;
    margin: 0;
  }

  .small-text {
    font-family: $lato;
    font-size: 15px;
    letter-spacing: 0.02em;
    padding: 10px;
  }

  .curve {
    width: 100%;
  }

  .image {
    padding: 20px 0px;
    text-align: center;
  }
}
