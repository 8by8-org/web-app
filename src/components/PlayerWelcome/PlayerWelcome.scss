@import "./../../styles/Components.scss";

.player-welcome {
  background-color: $white;
  color: $black;

  button {
    @include button;
    outline: 4px solid white;
    margin: 10px;
    margin-top: 20px;
  }

  .top {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    padding: 25px;
    background-color: $black;
    color: white;
    .top-heading {
      @include heading-text(white, 50px);
      line-height: 57.6px;
      letter-spacing: 0.03em;
      margin: 20px 0px 20px;
      .underline {
        @include text-underline;
      }
    }

    a:link,
    a:active,
    a:visited {
      color: $blue;
    }
  }

  a:link,
  a:active,
  a:visited {
    color: $black;
  }

  .main-content {
    padding: 20px 40px;
  }

  .heading {
    @include heading-text($black, 36px);
    margin-top: 25px;
    margin-bottom: 20px;
    line-height: 43.2px;
    letter-spacing: 0.03em;
    text-align: center;
    .underline {
      @include text-underline;
    }
  }

  .subheading {
    @include subheading-text($black, 24px);
    line-height: 29px;
    padding: 0px;
  }

  .text {
    font-family: $lato;
    font-size: 18px;
    line-height: 22px;
    letter-spacing: 0.02em;
    margin: 0;
  }

  .small-text {
    font-family: $lato;
    font-size: 15px;
    letter-spacing: 0.02em;
    padding: 10px;
  }

  .curve {
    width: 100%;
  }

  .image {
    padding: 20px;
    padding-bottom: 60px;
    text-align: center;
  }
}
