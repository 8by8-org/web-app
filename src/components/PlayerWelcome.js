import { useHistory } from 'react-router-dom'
import SignUp1 from "../assets/images/PlayerWelcome/SignUp1.png"
import SignUp2 from "../assets/images/PlayerWelcome/SignUp2.png"
import Vote from "../assets/images/PlayerWelcome/Vote1.png"
import Calendar from "../assets/images/PlayerWelcome/Complete1.svg"
import Top from '../assets/images/PlayerWelcome/Top.svg'
import './PlayerWelcome.scss'

export default function PlayerWelcome() {
    const history = useHistory(); 

    return (
        <div className="player-welcome">
            <div className="top" align="center">
                <img src={Top}/>
            </div>
            <div className="main-content">
                <div>
                    <h2 className="heading"><u className="underline">Support</u> Yang's 8by8 Challenge!</h2>
                    <div align="center">
                        <img src={Calendar} />
                    </div>
                </div>
                <div className="text">
                    <p>
                        <b>
                            Help "challenger" win their <u>8BY8 Challenge</u> by registering to vote or taking other ctions to #stopasianhate!
                        </b>
                    </p>
                </div>
                <div>
                    <button onClick={() => {history.push("/actions")}}>Get Started</button>
                </div>
                <div align="center">
                    <p className="small-text">Already have an account? Sign In</p>
                </div>
                <div>
                    <h3 className="heading"><u className="underline">Here's how it works</u></h3>
                </div>
                <p className="subheading">
                    1. ALL WE NEED IS YOUR ACTION
                </p>
                <p className="text">
                    You can take any number of the available actions: register to vote, get election reminders, 
                    or take the 8by8 challenge yourself. Pick one to start.
                </p>
                <div className="image">
                        <img src={SignUp1} alt="8by8 Logo" />
                </div>
                <p className="subheading">
                    2. YOUR FRIEND WILL EARN A BADGE
                </p>
                <p className="text">
                    Any of the 3 actions will help your friend earn a badge, and get closer to winning the challenge.
                </p>
                <div className="image">
                        <img src={SignUp2} alt="8by8 Logo" />
                </div>
                <p className="subheading">
                    3. COME BACK AND TAKE MORE ACTIONS
                </p>
                <p className="text">
                    Whether it is to help the same friend or a different one, the more action you take, the better! 
                    Note that you can only help earn on badge per friend.     
                </p>
                <div className="image">
                        <img src={Vote} alt="8by8 Logo" />
                </div>
                <button onClick={() => {history.push("/actions")}}>Get Started</button>
                    <p className="small-text" align="center">Already have an account? Sign In</p>
            </div>
            </div>
    )
}
