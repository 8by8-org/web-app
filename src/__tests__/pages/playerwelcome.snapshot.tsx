import PlayerWelcome from '@/pages/player-welcome';
import { cleanup, render } from '@testing-library/react';

import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))

describe('PlayerWelcome', () => {
  afterEach(cleanup);
  
  it('renders player-welcome page unchanged', () => {
    const { container } = render(<PlayerWelcome />);
    expect(container).toMatchSnapshot();
  });
});