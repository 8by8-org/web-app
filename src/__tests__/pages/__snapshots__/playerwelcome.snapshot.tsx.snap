// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerWelcome renders player-welcome page unchanged 1`] = `
<div>
  <div
    class="outer_container"
  >
    <div
      class="inner_container"
    >
      <section
        class="section_1"
      >
        <h1
          class="top_header"
        >
          <u
            class="underline"
          >
            Support
          </u>
           
          the
           
          8by8 Challenge!
        </h1>
        <p>
          <b
            class="content"
          >
            Help
             
            your friend
             
            win their 
            <u>
              8by8 Challenge
            </u>
             by registering to vote or taking other actions to #stopasianhate!
          </b>
        </p>
        <button
          class="get_started_btn"
          type="button"
        >
          Get Started
        </button>
        <p
          class="signin_line"
        >
          Already have an account?
           
          <a
            class="teal_link"
            href="/signin"
          >
            Sign In
          </a>
        </p>
        <a
          class="teal_link"
          href="/why8by8"
        >
          See why others are doing it
        </a>
      </section>
      <div
        class="curve_container"
      >
        <img
          alt="Black Curve"
          class="black_curve"
          data-nimg="1"
          decoding="async"
          height="40"
          id="black_curve"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <section
        class="section_2"
      >
        <h2
          class="underline"
        >
          Here's how it works
        </h2>
        <h3
          class="step_header"
        >
          1. Choose an action to take
        </h3>
        <p
          class="step_text"
        >
          You can take any number of the available actions: register to vote, get election reminders or take the 8by8 challenge yourself. Pick one to start.
        </p>
        <img
          alt="Take Action"
          class="image"
          data-nimg="1"
          decoding="async"
          height="40"
          id="take_action"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <h3
          class="step_header"
        >
          2. Your friend will earn a badge
        </h3>
        <p
          class="step_text"
        >
          Any of the 3 actions will help your friend earn a badge and get closer to winning the challenge.
        </p>
        <img
          alt="Friend Earns a Badge"
          class="image"
          data-nimg="1"
          decoding="async"
          height="40"
          id="earn_badge"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <h3
          class="step_header"
        >
          3. Come back and take more actions
        </h3>
        <p
          class="step_text"
        >
          Whether it is to help the same friend or a different one, the more actions you take, the better! Note that you can only help earn one badge per friend.
        </p>
        <img
          alt="More Actions"
          class="image"
          data-nimg="1"
          decoding="async"
          height="40"
          id="more_actions"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <h3
          class="step_header"
        >
          4. Get a Reward!
        </h3>
        <p
          class="step_text"
        >
          When your friend wins the challenge within 8 days, you win as well! Then select and enjoy a reward from one of our amazing partners.
        </p>
        <img
          alt="Get a Reward"
          class="image"
          data-nimg="1"
          decoding="async"
          height="40"
          id="get_reward"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <button
          class="get_started_btn"
          type="button"
        >
          Get Started
        </button>
        <p
          class="signin_line"
        >
          Already have an account?
           
          <a
            class="signin_link_black"
            href="/signin"
          >
            Sign In
          </a>
        </p>
      </section>
    </div>
  </div>
</div>
`;
